fecgov/openFEC
# Fix npm warnings in circleci builds

## Summary (required)

NPM warnings appear when circleci runs  during `Install node dependencies`,  `Deploy API` tasks. The warnings indicated the Node/NPM versions are unstable during deployments.

 After upgrading Node to v14.15.5 and NPM to v8.3.1 ,the circleci builds run OK without any warnings.


- Resolves #5021 

1. Updated Node to v14.15.5 and NPM to v8.3.1 in package.json file
2. Update Node to v14.15.5 in circleci/config.yml file

### Required reviewers

Two developers

## Impacted areas of the application

-  circleci builds

## Screenshots

Before:
<img width="1102" alt="Screen Shot 2022-01-14 at 1 05 53 PM" src="https://user-images.githubusercontent.com/11650355/149563922-357ef63e-8a3f-40b2-8538-b6a72c2133c6.png">


After:
<img width="1118" alt="Screen Shot 2022-01-14 at 1 06 31 PM" src="https://user-images.githubusercontent.com/11650355/149563910-e714c104-39ac-41b1-814e-27c3961ddc58.png">


## How to test

-  checkout this branch
- remove local node_modules folder : `rm -rf node_modules`
- delete package-lock.json file
- `nvm install v14.15.5`
- `nvm alias default v14.15.5`
- `nvm use default`
- `npm install -g npm` (npm@8.3.1 will be installed)
- `npm i`
- `npm run build`
- `./manage.py runserver` (make sure local API runs OK)
- rerun or deploy this branch to `dev` space to verify that the NPM warnings no longer appear in the `Install node dependencies` or `Deploy API` circleci tasks.

Circleci build:
https://app.circleci.com/pipelines/github/fecgov/openFEC/1198/workflows/d05c9446-447a-4c09-af4a-b39efbd32589/jobs/3692
